include/deal.II/lac/precondition.h

```diff
@@ -1472,13 +1472,16 @@ PreconditionChebyshev<MATRIX,VECTOR>::initialize (const MATRIX &matrix,
               ExcMessage ("Need to set at least two iterations to find eigenvalues."));
 
       // set a very strict tolerance to force at least two iterations
-      ReductionControl control (data.eig_cg_n_iterations, 1e-20, 1e-20);
+      ReductionControl control (data.eig_cg_n_iterations, 1e-35, 1e-10);
       GrowingVectorMemory<VECTOR> memory;
       VECTOR *rhs = memory.alloc();
       VECTOR *dummy = memory.alloc();
-      rhs->reinit(data.matrix_diagonal_inverse, true);
+      rhs->reinit(data.matrix_diagonal_inverse);
       dummy->reinit(data.matrix_diagonal_inverse);
       *rhs = 1./std::sqrt(static_cast<double>(matrix.m()));
+      if (rhs->locally_owned_elements().is_element(0))
```

Comment:
You want to avoid putting a plain constant vector on the right hand side (but on the other hand a vector close to constant gives the largest eigenvalue estimate reasonably quickly). If you work with a purely Neumann constrained problem and fix the constant by projecting out the constant mode in your matrix and right hand side, this right hand side makes trouble (because you do not know of a projection a user might have applied to the rhs).

In the longer run, I would like to allow for computing the eigenvalue the first time the preconditioner is applied, but that makes the preconditioner variable and one needs to be careful.